app/controllers/items_controller.rb

```diff
@@ -2,18 +2,86 @@ class ItemsController < OauthController
   before_action :login
 
   def refile
+    @refile_request = RefileRequest.new
+    # TODO: Finish this in subsequent branch.
+    if params[:page] && params[:per_page]
+      @offset = ( params[:page].to_i - 1 ) * params[:per_page].to_i
+    else
+      @offset = 0
+    end
+
+    date_start = params[:date_start].present? ? params[:date_start] : nil
+    date_end = params[:date_end].present? ? params[:date_end] : nil
+    @refiles = get_refiles(date_start,date_end,params[:page],params[:per_page])
   end
   
   def update_metadata
+    @barcodes = params[:notice].present? && params[:notice]["barcodes"].present? ? params[:notice]["barcodes"] : []
+    @protect_cgd = params[:notice].present? && params[:notice]["protect_cgd"].present? ? params[:notice]["protect_cgd"] : false
   end
 
   def send_metadata
-    barcodes = params[:barcodes]
```

Comment:
We talked about not needing an `invalid_barcodes` local variable because we can rely on `Message#invalid_barcodes`. We may want to add a few more specs to the `Message` unit tests about how `invalid_barcodes`'s state starts and changes.